src/Command/RunCommand.php

```diff
@@ -70,6 +75,82 @@ public function __construct(
         $this->legacyOauthApiUrl = $legacyOauthApiUrl;
         $this->instanceLimits = $instanceLimits;
         $this->logProcessor = $logProcessor;
+
+        pcntl_signal(SIGTERM, [$this, 'sigHandler']);
+        pcntl_signal_dispatch();
+        pcntl_async_signals(true);
+
+    }
+
+    public function sigHandler(int $signo, $siginfo): void
```

Comment:
Hele, uprimne jsem o tom mel uplne jinou predstavu.. misto toho, aby ten signal handler fakticky neco terminoval, bych jen pri prijeti signalu nastavil nejaky priznak `private bool $terminationRequested` a ten kontroloval prubezne v `execute`.  Ale uz chapu, ze tam bude problem s tim, ze `$runner->run()` je blokujici.

Ten `run` by se musel unwrapnout, aby loop cekajici na dokonceni jobu byla tady v tom commandu. Jinak se ty race condition asi nezbavime, no.

Nebo anopak predavat dal, dovnitr Runner, info o prijeti signalu k terminaci a resit tu logiku tam, ale to by bylo IMHO o kus komplikovanejsi.